substrate/pkg/controller/substrate/infrastructure/subnets.go

```diff
@@ -97,12 +97,21 @@ func (s *Subnets) ensureSubnet(ctx context.Context, substrate *v1alpha1.Substrat
 	if len(describeSubnetsOutput.Subnets) > 0 {
 		logging.FromContext(ctx).Infof("Found subnet %s", aws.StringValue(name))
 		return describeSubnetsOutput.Subnets[0], nil
+
+	}
+	// tag required by ELB controller to discover these subnets to configure ELB
+	tags := []*ec2.Tag{
+		{Key: aws.String("kubernetes.io/role/elb"), Value: aws.String("1")},
+	}
+	if subnetSpec.Public {
+		// these tags are required by Karpenter controller to discover these subnets to provision new node
+		tags = append(tags, &ec2.Tag{Key: aws.String("karpenter.sh/discovery"), Value: aws.String(substrate.Name)})
 	}
 	createSubnetsOutput, err := s.EC2.CreateSubnetWithContext(ctx, &ec2.CreateSubnetInput{
 		AvailabilityZone:  aws.String(subnetSpec.Zone),
 		CidrBlock:         aws.String(subnetSpec.CIDR),
 		VpcId:             substrate.Status.Infrastructure.VPCID,
-		TagSpecifications: discovery.Tags(substrate, ec2.ResourceTypeSubnet, name),
+		TagSpecifications: discovery.Tags(substrate, ec2.ResourceTypeSubnet, name, tags...),
```

Comment:
I think it's more idiomatic to do
```
append(discovery.Tags(...), tags...)
```

Reply:
Answered above where the function is defined.